thetis/options.py

```diff
@@ -119,56 +144,106 @@ class ExplicitTimestepperOptions2d(ExplicitTimestepperOptions):
         'sub_pc_type': 'ilu',
         'mat_type': 'aij',
     }).tag(config=True)
-    solver_parameters_tracer = PETScSolverParameters({
+
+
+class ExplicitTracerTimeStepperOptions2d(ExplicitTimeStepperOptions2d):
+    """
+    Options for 2d explicit time integrator
+    applied to tracer equations
+    """
+    solver_parameters = PETScSolverParameters({
         'ksp_type': 'gmres',
         'pc_type': 'sor',
     }).tag(config=True)
 
 
-class IMEXTimestepperOptions2d(SemiImplicitTimestepperOptions2d):
-    """Options for 2d implicit-explicit time integrators"""
-    # TODO: Meaningful solver parameters
+class IMEXSWETimeStepperOptions2d(SemiImplicitTimeStepperOptions2d):
+    """
+    Options for 2d implicit-explicit time integrators
+    applied to shallow water equations
+    """
+    # TODO: Meaningful solver parameters for DIRK
     solver_parameters = PETScSolverParameters({
         'ksp_type': 'gmres',
         'pc_type': 'fieldsplit',
         'pc_fieldsplit_type': 'multiplicative',
     }).tag(config=True)
-    solver_parameters_dirk = PETScSolverParameters({
-        'ksp_type': 'gmres',
-        'pc_type': 'fieldsplit',
-        'pc_fieldsplit_type': 'multiplicative',
-    }).tag(config=True)
 
 
-class ExplicitTimestepperOptions3d(ExplicitTimestepperOptions):
-    """Base class for all 3d time stepper options"""
-    solver_parameters_2d_swe = PETScSolverParameters({
+class TimeStepperOptions3d(TimeStepperOptions):
+    """Options for 3d explicit time integrator"""
```

Comment:
`TimeStepperOptions3d` inherits empty `solver_parameters` dict from `TimeStepperOptions`. This is a bit confusing as it is not used (could actually cause bugs). Maybe better to inherit from `FrozenHasTraits` directly? The reason for the differentiation is that `TimeStepperOptions3d` is a _collection_ of timestepper options, not really an instance of that kind itself.